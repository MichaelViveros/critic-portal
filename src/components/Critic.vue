<template>
  <div>
    <img
      alt="Critic pic"
      :src="this.critic.imgSrc"
    />
    <h1>{{ critic.name }}</h1>
    <p>{{ critic.bio }}</p>
    <p>Random Fact:</p>
    <p>{{ fact }}</p>
    <button @click="changeFact">New Random Fact</button>
  </div>
</template>

<script>
export default {
  name: 'Critic',
  props: {
    critic: Object,
  },
  computed: {
    fact: function () {
      return this.critic.randomFacts[this.factIndex]
    },
  },
  data: function () {
    return {
      factIndex: 0,
    }
  },
  methods: {
    changeFact: function () {
      const max = this.critic.randomFacts.length
      this.factIndex = Math.floor(Math.random() * max)
    }
  }
}
</script>

<style scoped>
</style>
