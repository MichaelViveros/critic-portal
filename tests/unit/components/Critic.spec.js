import { mount } from '@vue/test-utils'
import Critic from '@/components/Critic.vue'

describe('Critic.vue', () => {

  let propsData

  beforeEach(() => {
    propsData = {
      critic: {
        name: "Justin Chang",
        bio: "Chief film critic for the LA Times.",
        imagePath: "../assets/justin-chang.jpeg",
        randomFacts: [
          "Had a line in Minari",
          "Has met Michael Viveros",
          "Ryan Coogler is a big fan",
          "Favourite movie is Chunking Express"
        ],
      }
    }
  })
  
  it('renders correctly', () => {
    const wrapper = mount(Critic, { propsData })
    expect(wrapper).toMatchSnapshot(); 
  })

  it('displays random fact when button is clicked', async () => {
    jest.spyOn(global.Math, 'random').mockReturnValue(0.5);
    const wrapper = mount(Critic, { propsData })
    expect(wrapper.text()).toContain(propsData.critic.randomFacts[0])
    await wrapper.find('button').trigger('click')
    // 0.5 * 4 = 2, fact with index 2 is 'randomly' chosen
    expect(wrapper.text()).toContain(propsData.critic.randomFacts[2])
  })
})
