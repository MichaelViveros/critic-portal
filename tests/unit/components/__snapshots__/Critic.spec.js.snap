// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Critic.vue renders correctly 1`] = `
<div><img alt="Critic pic">
  <h1>Justin Chang</h1>
  <p>Chief film critic for the LA Times.</p>
  <p>Random Fact:</p>
  <p>Had a line in Minari</p> <button>New Random Fact</button>
</div>
`;
